lib/sord/rbi_generator.rb

```diff
@@ -176,106 +111,100 @@ def add_methods(item, indent_level)
-            name = name.chop if name.end_with?(':')
-            "#{name}: #{inferred_type}"
+            Logging.infer("inferred type of parameter #{name.inspect} as #{inferred_type} using getter's return type", meth)
+            inferred_type
           else
             # Is this the only argument, and was a @param specified without an
             # argument name? If so, infer it
-            if parameter_names_to_tags.length == 1 \
+            if parameter_names_and_defaults_to_tags.length == 1 \
               && meth.tags('param').length == 1 \
               && meth.tag('param').types
 
-              Logging.infer("argument name in single @param inferred as #{parameter_names_to_tags.first.first.inspect}", meth, indent_level)
-              "#{name}: #{TypeConverter.yard_to_sorbet(meth.tag('param').types, meth, indent_level, @replace_errors_with_untyped, @replace_unresolved_with_untyped)}"
+              Logging.infer("argument name in single @param inferred as #{parameter_names_and_defaults_to_tags.first.first.first.inspect}", meth)
+              TypeConverter.yard_to_sorbet(meth.tag('param').types, meth, @replace_errors_with_untyped, @replace_unresolved_with_untyped)
             else
-              Logging.omit("no YARD type given for #{name.inspect}, using T.untyped", meth, indent_level)
-              # Get rid of : on keyword arguments.
-              name = name.chop if name.end_with?(':')
-              "#{name}: T.untyped"
+              Logging.omit("no YARD type given for #{name.inspect}, using T.untyped", meth)
+              'T.untyped'
             end
           end
         end
 
         return_tags = meth.tags('return')
         returns = if return_tags.length == 0
-          Logging.omit("no YARD return type given, using T.untyped", meth, indent_level)
-          "returns(T.untyped)"
+          Logging.omit("no YARD return type given, using T.untyped", meth)
+          'T.untyped'
         elsif return_tags.length == 1 && return_tags&.first&.types&.first&.downcase == "void"
-          "void"
+          nil
         else
-          "returns(#{TypeConverter.yard_to_sorbet(meth.tag('return').types, meth, indent_level, @replace_errors_with_untyped, @replace_unresolved_with_untyped)})"
+          TypeConverter.yard_to_sorbet(meth.tag('return').types, meth, @replace_errors_with_untyped, @replace_unresolved_with_untyped)
         end
 
-        prefix = meth.scope == :class ? 'self.' : ''
-
-        add_signature(sig_params_list, returns, indent_level)
+        parlour_params = parameter_names_and_defaults_to_tags
+          .zip(parameter_types)
+          .map do |((name, default), _), type|
+            Parlour::RbiGenerator::Parameter.new(name.to_s, type: type, default: default)
+          end
 
-        rbi_contents << "#{'  ' * (indent_level + 1)}def #{prefix}#{meth.name}(#{parameter_list}); end"
+        @current_object.create_method(meth.name.to_s, parlour_params, returns, class_method: meth.scope == :class)
```

Comment:
Perhaps a bit late to suggest this - and not really a problem with Sord - but the parameters should really use named parameters. Right now it's not super clear what's being passed in/why.

e.g.:
```ruby
@current_object.create_method(method_name: meth.name.to_s, params: parlour_params, returns: returns, class_method: meth.scope == :class)
```

Reply:
This certainly helps readability, and if library changes are going to be made to Parlour, they may as well be made early on. I'll work on this!